File: 1544-make-the-string-great/1544-make-the-string-great.java
class Solution {
    public String makeGood(String s) {
         Stack<Character> st = new Stack<>();
        
        for(int i=0;i<s.length();i++)
        {
            if(st.empty())
            {
                st.push(s.charAt(i));
            }
            else if(st.peek()==s.charAt(i)+32 || st.peek()+32==s.charAt(i))
                st.pop();
            else
                st.push(s.charAt(i));
        }
        
        StringBuilder res=new StringBuilder();
        while(!st.empty())
        {
            res.append(st.pop());
        }
         return res.reverse().toString();
    }
}

File: 1544-make-the-string-great/1544-make-the-string-great.cpp
class Solution {
public:
    string makeGood(string s) {
        stack<char>st;
        for(int i=0;i<s.size();i++)
        {
            if(st.empty())
            st.push(s[i]);
            else if(st.top()==s[i]+32 || st.top()+32==s[i])
             st.pop();
              else
                st.push(s[i]);
        }
        s="";
        while(!st.empty())
        {
            s.push_back(st.top());
            st.pop();
        }
        reverse(s.begin(),s.end());
        return s;
    }
};